conda/cli/conda_argparse.py

```diff
@@ -199,6 +225,34 @@ def _check_value(self, action, value):
         else:
             super()._check_value(action, value)
 
+    def parse_args(self, args=None, namespace=None):
+        """
+        We override this method to check if we are running from a known plugin subcommand.
+        If we are, we do not want to handle argument parsing as this is delegated to the plugin
+        subcommand. We instead return a ``Namespace`` object with ``plugin_subcommand`` defined,
+        which is a ``conda.plugins.CondaSubcommand`` object.
+        """
+        plugin_subcommand = None
+
+        if len(sys.argv) > 1:
+            name = sys.argv[1]
```

Comment:
Both of the following are possible and succeed:

```
conda --debug info
conda info --debug
```

So we shouldn't be exclusively checking the second argument. Is this something we should address here or deal with later?

Reply:
@kenodegard, thanks for bring this up. I think we don't have to worry about this for now.

Even if you tried to override the `info` command, it still wouldn't succeed in either case because we would either be checking the sub-command against "info" (which it would recognize as a built-in) or checking to see if `--debug` was a valid sub-command name.

Maybe you could elaborate a little more on exactly what you see as the risk here or how this would break things.

Reply:
The `info` subcommand was just an example, any subcommand can be run with certain flags present to the left of the subcommand:

```
conda --debug create
conda --debug config
conda --debug install
conda --debug search
```

Even the old way of invoking plugins would work:

```
conda --debug build
```

But this implementation wouldn't work whenever conda-build is refactored to use the new plugin framework since we are only looking at the second argument.

Reply:
Fair point. Because this currently doesn't break any of that behavior, I believe it is okay to do nothing for now. When we finally do decide to remove that old behavior of invoking sub-commands, we need to make sure that it accounts for this.